$NetBSD: patch-CMakeLists.txt,v 1.2 2019/01/27 17:04:16 nia Exp $

Avoid checking for a C++ compiler, since C++ isn't required for a
libretro build.

vfs-dirent peeks into non-standard structures and also isn't required
for a libretro build.

Fix locale detection.

--- CMakeLists.txt.orig	2019-01-27 00:06:06.000000000 +0000
+++ CMakeLists.txt
@@ -5,7 +5,7 @@ endif()
 if(POLICY CMP0072)
 	cmake_policy(SET CMP0072 NEW)
 endif()
-project(mGBA)
+project(mGBA C)
 set(BINARY_NAME mgba CACHE INTERNAL "Name of output binaries")
 if(NOT MSVC)
 	set(GCC_STD "c99")
@@ -245,7 +245,7 @@ elseif(UNIX)
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
 	endif()
 
-	list(APPEND CORE_VFS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/util/vfs/vfs-fd.c ${CMAKE_CURRENT_SOURCE_DIR}/src/util/vfs/vfs-dirent.c)
+	list(APPEND CORE_VFS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/util/vfs/vfs-fd.c)
 	file(GLOB OS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/platform/posix/*.c)
 	source_group("POSIX-specific code" FILES ${OS_SRC})
 endif()
@@ -391,7 +391,9 @@ if(HAVE_LOCALTIME_R)
 	list(APPEND FUNCTION_DEFINES HAVE_LOCALTIME_R)
 endif()
 
-if(HAVE_NEWLOCALE AND HAVE_FREELOCALE AND HAVE_USELOCALE OR APPLE)
+check_include_files("locale.h" HAVE_LOCALE)
+
+if(HAVE_LOCALE OR APPLE)
 	list(APPEND FUNCTION_DEFINES HAVE_LOCALE)
 	if (HAVE_SNPRINTF_L)
 		list(APPEND FUNCTION_DEFINES HAVE_SNPRINTF_L)
